mlflow/tensorflow/__init__.py

```diff
@@ -709,7 +748,9 @@ def _load_pyfunc(path):
         should_compile = save_format == "tf"
         K = importlib.import_module(keras_module.__name__ + ".backend")
         if K.backend() == "tensorflow":
-            K.set_learning_phase(0)
+            # This API doesn't exist for tf > 2.15.0
+            if Version(tensorflow.__version__) <= Version("2.15.0"):
```

Comment:
I would suggest using `hasattr(K, "set_learning_phase")`,

tensorflow might release version like `2.15.1` in future, it will break the condition

Reply:
We can't now because nightly version is '2.16.0-dev..' which is also < Version("2.16")

Reply:
We can use:
`if not Version(tensorflow.__version__).is_devrelease and Version(tensorflow.__version__) < Version("2.16"):`